models/user.go

```diff
@@ -2,22 +2,45 @@ package models
 
 import (
 	"fmt"
+
+	"github.com/wantedly/developers-account-mapper/services"
 )
 
-// User stores login user name and GitHub username
+// User stores login user name and accounts information
 type User struct {
-	LoginName string
+	LoginName      string
 	GitHubUsername string
+	SlackUsername  string
+	SlackUserId    string
 }
 
 // NewUser creates new User instance
-func NewUser(loginName string, githubUsername string) *User {
+func NewUser(loginName string, githubUsername string, slackUsername string, slackUserId string) *User {
 	return &User{
 		LoginName: loginName,
 		GitHubUsername: githubUsername,
+		SlackUsername: slackUsername,
+		SlackUserId: slackUserId,
 	}
 }
 
+func (u *User) SetSlackUserId(newId string) *User {
+	u.SlackUserId = newId
+	return u
+}
+
+func (u *User) RetrieveSlackUserId() (*User, error) {
+	nameIdMap, err := services.SlackUserList()
+	if err != nil {
+		return nil, err
+	}
+	u.SetSlackUserId(nameIdMap[u.SlackUsername])
+	return u, nil
+}
+
 func (u *User) String() string {
-	return fmt.Sprintf("%v:@%v", u.LoginName, u.GitHubUsername)
+	if u.SlackUserId == "" {
+		u.RetrieveSlackUserId()
```

Comment:
String is implicitly called here
`fmt.Printf("user %q added.\n", user)`
And, this seems to cause
```console
potsbo@magi-INS-% make && envchain wtd ./bin/developers-account-mapper register potsbo potsbo shimpei
go build -ldflags="-s -w -X \"main.Version=0.1.0\" -X \"main.Revision=8c5a955\"" -o bin/developers-account-mapper
# github.com/wantedly/developers-account-mapper/models
../../../../.go/src/github.com/wantedly/developers-account-mapper/models/user.go:40: cannot use nil as type string in return argument
make: *** [bin/developers-account-mapper] Error 2
```

I'm not 100% sure, though, maybe String should not return anything but a `string`.
Or, do you think we should assign like doing `str, err := user.String` and handle it?